lib/helpers.ts

```diff
@@ -209,23 +210,61 @@ export function parseRawAssayType(t:string) {
     return { name: t, level: null };
 }
 
-export function encodeSelectedFilters(selectedFilters:ExploreSelectedFilter[]) {
+export function urlEncodeSelectedFilters(selectedFilters:ExploreSelectedFilter[]) {
     return JSON.stringify(selectedFilters);
 }
-export function parseSelectedFilters(selectedFiltersURLQueryParam:string|undefined):ExploreSelectedFilter[] | null {
+export function parseSelectedFiltersFromUrl(selectedFiltersURLQueryParam:string|undefined):ExploreSelectedFilter[] | null {
     if (selectedFiltersURLQueryParam) {
         return JSON.parse(selectedFiltersURLQueryParam);
     }
     return null;
 }
 
+function addQueryStringToURL(url:string, queryParams: { [key:string]:string|undefined }) {
```

Comment:
@alisman as the URLWrapper mastermind, what do you think about all this url-handling code? feels a bit ad-hoc to me but I don't know if it's time to do it better yet

Reply:
seems fine for now.  but where is actual URL encoding take place?